### PR TITLE
Add docker file.

### PR DESCRIPTION
**Pull request by myidpt**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      iptables \
+      iproute2 \
+      iputils-ping \
+      knot-dnsutils \
+      netcat \
+      tcpdump \
+      net-tools \
+      lsof \
+      linux-tools-generic \
+      sudo \
```
Comment (evan2645) [inline docker/Dockerfile:24]: We need all this stuff? Some of it seems dangerous

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.PHONY: ifs docker clean
+
+ifs: *.go
+	go build -o docker ./...
```
Comment (evan2645) [inline Makefile]:
```suggestion
	go build -o istio-federation-server
```

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
+FROM ubuntu:bionic
```
Comment (evan2645) [inline docker/Dockerfile:1]: I think this file can live in the top level directory, instead of putting it in docker/

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+docker:
+	docker build -t istio/ifs docker
```
Comment (evan2645) [inline Makefile]: If we take the suggestion to move dockerfile to toplevel

```suggestion
	docker build -t istio/ifs .
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+clean:
+	rm docker/istio-federation-server
```
Comment (evan2645) [inline Makefile]: If we take the suggestion to move dockerfile to toplevel

```suggestion
	rm istio-federation-server
```

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,10 @@
+.PHONY: ifs docker clean
```
Comment (evan2645) [inline Makefile]: I think that the `ifs` target is not a phony one? It references the go files

```suggestion
.PHONY: docker clean
```

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+
+ifs: *.go
+	go build -o docker ./...
+
+docker:
```
Comment (evan2645) [inline Makefile]: Would be good for this to depend on the `ifs` target to make sure that it's actually built

```suggestion
docker: ifs
```

**Review by evan2645 — APPROVED** [review]
Thanks @myidpt!

I left a handful of small comments that don't matter _too_ much but will make for a slightly better experience. I left GitHub suggestions wherever I could to keep the overhead of accepting them as low as possible. I realize that this is probably just copy pasta, so no worries if you want to punt on these changes for now... let me know how you'd like to proceed

File: Makefile (hunk 6 above)
Comment (myidpt) [reply Makefile]: Yes I just added it without thinking. I can remove it.

File: docker/Dockerfile (hunk 1 above)
Comment (myidpt) [reply docker/Dockerfile:24]: Yeah, I just put them here for convenience... I'll remove some. Let's not worry about it too much for now :)

File: docker/Dockerfile (hunk 3 above)
Comment (myidpt) [reply docker/Dockerfile:1]: Sure. I'll move it.

**Comment by myidpt** [comment]
I think this PR is ready. Please feel free to merge. I don't have write access.

**Comment by evan2645** [comment]
> I don't have write access.

I listed you as a collaborator, but when I look at the page it says that you need to accept the invitation. I think they usually send an email, but in the event you didn't get it you should be able to use the link below to confirm... Hopefully you should have write access after that

https://github.com/evan2645/istio-federation-server/invitations

**Pull request merged**